pandas/core/indexes/datetimes.py

```diff
@@ -2040,7 +2040,11 @@ def date(self):
         Returns numpy array of python datetime.date objects (namely, the date
         part of Timestamps without timezone information).
         """
-        return libts.ints_to_pydatetime(self.normalize().asi8, box="date")
+        if (self.tz is None):
+            return libts.ints_to_pydatetime(self.normalize().asi8, box="date")
```

Comment:
Do we need the `.normalize()` calls here? I think `ints_to_pydatetime` will strip the time information away when constructing the date